parquet-common/src/main/java/org/apache/parquet/io/DelegatingPositionOutputStream.java

```diff
@@ -35,7 +35,11 @@ public OutputStream getStream() {
 
   @Override
   public void close() throws IOException {
-    stream.close();
+    try (OutputStream os = this.stream) {
```

Comment:
I also suspect that delegate classes like `LittleEndianDataOutputStream` and `DelegatingPositionOutputStream` should take this complexity.

Reply:
Sorry I did not get what you mean.. Are you saying that these classes should not be bothered with flush etc?

Reply:
Yes, for example `LittleEndianDataOutputStream` is used by column encoders which is accessed via `ValuesWriter.getBytes()` at https://github.com/apache/parquet-mr/blob/master/parquet-column/src/main/java/org/apache/parquet/column/values/ValuesWriter.java#L43. We can see that `getBytes()` implementations have guaranteed to call flush() before consuming the encoded bytes. We don't have to call flush() in the close() since the data is not used any more.

Reply:
I see. well if you all think so, can remove that .. Though the field var is defined as OutputStream, so in theory could contain a buffered output stream or any other custom OutputStream with some buffered data and may or may not have flush implemented or may or may not have flush invoked before close.

Reply:
This is not a strong opinion.